packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,401 @@
+    of the comments feature. Once the visual design for these components is complete, we will then be implementing those specific changes within the defined
+    scope of development. However, we will still make use of existing nimble components such as `nimble-button` and `nimble-text-area` to maintain a
+    consistent design for the initial release.
+
+### Prior Art/Examples
+
+Sample Mockup Screenshots
+
+_`nimble-rich-text-editor`_
+
+![Editor](./spec-images/editor-sample.png)
+
+_`nimble-rich-text-viewer`_
+
+![Viewer](./spec-images/viewer-sample.png)
+
+## Design
+
+### `nimble-rich-text-editor`
+
+The `nimble-rich-text-editor` will be divided into two sections namely an `editor` section and a `footer` section.
+
+1. `editor` section is the actual text area to add or update rich text content.
+2. `footer` section consists of `nimble-button` to control each text formatting functionalities like bold, italic, etc, and a
+   `footer-actions` slot element which is typically used to add action buttons to the right bottom of the component.
+
+Example usage of the `nimble-rich-text-editor` in the application layer is as follows:
+
+```html
+<nimble-rich-text-editor>
+    <div slot="footer-actions">
+        <nimble-button>Cancel</nimble-button>
+        <nimble-button>Add Comment</nimble-button>
+    </div>
+</nimble-rich-text-editor>
+```
+
+### API
+
+_Props/Attrs_
+
+-   `markdownValue` - is an accessor used to get and set the markdown value.
+    -   `getter` - this will serialize the content by extracting the Node from the editor and convert it into a markdown string using
+        [prosemirror-markdown serializer](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/to_markdown.ts#L30).
+    -   `setter` - this will parse the markdown string into a Node and load it back into the editor using
+        [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/from_markdown.ts#L199).
+
+_Methods_
+
+-   none
+
+_Events_
+
+-   `input` - event emitted when there is a change in the editor. This can be achieved through Tiptap's [update event](https://tiptap.dev/api/events#update).
+    Below are a few scenarios to understand when this event will fire:
+
+    1. This event will fired for every input in the content of the editor, including text inputs, text formatting changes, and text removals.
+    2. This event will not fire when there are no changes made to the content of the editor. For example, all mouse events, selecting the texts, state
+       changes, etc,
+
+    The emitted event data is in the format of a `CustomEvent` and consists of a `detail` object with an `isEmpty` boolean flag indicating whether the editor
+    is empty or not.
+
+_CSS Classes and CSS Custom Properties that affect the component_
+
+-   The minimum and maximum height of the `editor section` will be fixed value. If the content exceeds the maximum height, we will show the vertical scrollbar to
```

Comment:
Thanks for the detailed information. I have updated the spec as per the suggestion. We will get started with this approach for the initial release and as said as per the visual design from Rick, we will incorporate the changes during the implementation or in the subsequent releases.

Reply:
Looking good, thanks for updating. A couple last minor points:

>The client will determine the default and maximum height and width of the component.

Nimble should provide a default height and width for the case where the element is placed on the page with no size styling. This can just be a reasonable fixed pixel size, as long as the client can override it.

Also, a possible nice-to-have feature for the editor component would be to use the [built-in resize attribute](https://developer.mozilla.org/en-US/docs/Web/CSS/resize) to allow the user to dynamically grow the height of the text input box (like GitHub allows). I bet it would be trivial to implement, but it's just an idea so no need to capture it if you're worried about scope.

![image](https://github.com/ni/nimble/assets/10500124/e4c135a7-94ed-4efa-afce-e005d6fa6a3c)

Reply:
Thanks for suggesting this @jattasNI. This new feature looks great. And from your reference, as you mentioned it seems easy to integrate. Can we please consider this after completing all the required features? As I am not sure of the development effort to write test scripts for this feature, it would be easy for us to analyze this once after making good progress on the development.

Reply:
Definitely, that makes sense.